### PR TITLE
PR: Adopt CI features from travertino

### PR DESCRIPTION
**Pull request by fanninpm**
Some "pybee" mentions were also replaced. I'm referencing the list found in #70.

### CODE REVIEW

**Comment by freakboy3742** [comment]
Hi @fanninpm - Just letting you know that I've seen this PR, and I'm really excited to see this work done. If you need an interim or final review, tag me and I'll take a look.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,21 +12,24 @@ jobs:
...
       matrix:
-        task: ['pycodestyle']
+        task:
+          - 'flake8'
+          - 'towncrier-check'
+          - 'package'
     steps:
     - uses: actions/checkout@v1
-    - name: Set up Python 3.5
+    - name: Set up Python 3.8
```
[the remark this answers is not in the document]
Comment (fanninpm) [reply .github/workflows/ci.yml]: Resolved as requested.

File: docs/tutorial/tutorial-1.rst [hunk 2]
```diff
@@ -13,7 +13,7 @@ apply them to any DOM-like tree of objects. There is no required base class;
 Colosseum will duck-type any object providing the required API. The simplest
 possible DOM node is the following::
 
-    from colosseum.dimensions import Size, Box
+    from src.colosseum.dimensions import Size, Box
```
[the remark this answers is not in the document]
Comment (fanninpm) [reply docs/tutorial/tutorial-1.rst]: Resolved as requested.

File: setup.cfg [hunk 3]
```diff
@@ -1,15 +1,66 @@
...
+ignore = E133,E226,W503
 
 [isort]
 combine_as_imports = true
 include_trailing_comma = true
 line_length = 79
 multi_line_output = 5
 not_skip = __init__.py
+
+[yapf]
```
[the remark this answers is not in the document]
Comment (fanninpm) [reply setup.cfg:63]: I used YAPF to fix the hundreds of spacing errors reported by `flake8` (hence why there are 600+ changed files in this PR). (I was originally going to use [`black`](https://github.com/psf/black) for this purpose, but `black` conflicts with this project's style guide.)

File: tests/engine/block_layout/test_block_non_replaced_normal_flow.py [hunk 4]
```diff
@@ -1,5 +1,5 @@
-from colosseum.constants import AUTO, BLOCK, RTL, SOLID
-from colosseum.declaration import CSS
+from src.colosseum.constants import AUTO, BLOCK, RTL, SOLID
+from src.colosseum.declaration import CSS
```
[the remark this answers is not in the document]
Comment (fanninpm) [reply tests/engine/block_layout/test_block_non_replaced_normal_flow.py]: Resolved as requested. (81a125d)

File: setup.py [hunk 5]
```diff
@@ -1,52 +1,16 @@
...
 import io
 import re
+from setuptools import setup
 
-from setuptools import find_packages, setup
+# Setuptools currently balks at reading the name
+# and version from setup.cfg.
 
-with io.open('./colosseum/__init__.py', encoding='utf8') as version_file:
+with io.open('./src/colosseum/__init__.py', encoding='utf8') as version_file:
```
[the remark this answers is not in the document]
Comment (fanninpm) [reply setup.py]: This is from before I upgraded to the latest version of `setuptools`, which fixes a problem that I encountered that prevented me from doing the elegant thing. Resolved as requested.

File: src/colosseum/engine.py [hunk 6]
```diff
@@ -177,25 +177,10 @@ def layout_box(display, node, containing_block, viewport, font):
...
-        node.layout.border_bottom_width = calculate_size(node.style.border_bottom_width, horizontal)
-
-    if node.style.border_left_style is None:
-        node.layout.border_left_width = 0
-    else:
-        node.layout.border_left_width = calculate_size(node.style.border_left_width, vertical)
+    node.layout.border_top_width = calculate_size(node.style.border_top_width, horizontal)
+    node.layout.border_right_width = calculate_size(node.style.border_right_width, vertical)
+    node.layout.border_bottom_width = calculate_size(node.style.border_bottom_width, horizontal)
+    node.layout.border_left_width = calculate_size(node.style.border_left_width, vertical)
```
[the remark this answers is not in the document]
Comment (fanninpm) [reply src/colosseum/engine.py:183]: Flake8 complained about the function being too complex. I could revert the change at the cost of flake8 failing once this PR is merged.

**Comment by freakboy3742** [comment]
@fanninpm Also: Confirming since you're evidently working on this right now, but haven't done so yet - there's a merge conflict with master. You'll need to resolve that conflict before CI will execute again.

File: tests/engine/block_layout/test_child_layout.py [hunk 7]
```diff
@@ -1,8 +1,9 @@
-from unittest import expectedFailure
+# from unittest import expectedFailure
```
[the remark this answers is not in the document]
Comment (fanninpm) [reply tests/engine/block_layout/test_child_layout.py]: Resolved as requested.

File: utils/w3c.py [hunk 8]
```diff
@@ -191,7 +191,7 @@ def startup(self):
 
         self.main_window.content = webview
 
-        files = []
+        # files = []
```
[the remark this answers is not in the document]
Comment (fanninpm) [reply utils/w3c.py]: Resolved as requested.

File: tests/test_declaration.py [hunk 9]
```diff
@@ -1,15 +1,15 @@
...
-from colosseum.declaration import CSS, validated_property
-from colosseum.units import percent, px
-from colosseum.validators import (is_color, is_integer, is_length, is_number,
-                                  is_percentage, is_uri)
-from colosseum.wrappers import BorderSpacing, Quotes
+from src.colosseum import engine as css_engine
+from src.colosseum.colors import GOLDENROD, NAMED_COLOR, REBECCAPURPLE
+from src.colosseum.constants import (AUTO, BLOCK, INHERIT, INITIAL, INLINE, LEFT,
+                                     REVERT, RIGHT, RTL, TABLE, UNSET, Choices,
+                                     OtherProperty)
```
[the remark this answers is not in the document]
Comment (fanninpm) [reply tests/test_declaration.py]: Resolved as requested.

File: setup.cfg [hunk 10]
```diff
@@ -1,15 +1,66 @@
...
+# E133: closing bracket missing indentation
+# E226: missing whitespace around arithmetic operator
+# W503: line break occurred before a binary operator
+ignore = E133,E226,W503
 
 [isort]
 combine_as_imports = true
 include_trailing_comma = true
 line_length = 79
 multi_line_output = 5
```
[the remark this answers is not in the document]
Comment (fanninpm) [reply setup.cfg]: This came from an early version of travertino's/briefcase's `setup.cfg`. Resolved as requested. (I think it'd also be a good idea to ignore the `.tox` directory.)

[the remark this answers is not in the document]
Comment (freakboy3742) [reply setup.cfg]: Completely agreed; the configuration for briefcase or toga might be a better reference point here, rather than Travertino.

**Comment by fanninpm** [comment]
Why did https://github.com/beeware/colosseum/runs/839908933 fail?

**Comment by freakboy3742** [comment]
@fanninpm Check the logs for CI. If you click on "Details" next to the failed test, you can get the full logs for what ran, and how it failed. You should be able to reproduce the same failure on your own machine by running the `package` tox environment (with `tox -e package`).

From the look of it, there's something wrong with the new `setup.cfg` configuration - the packaging process has created a wheel called `UNKNOWN-0.0.0-py3-none-any.whl`, which suggests it hasn't picked up either the version number or package name correctly.

**Comment by freakboy3742** [comment]
Another possible cause: if the test environment has an old version of `setuptools` or `wheel`. Older versions of those tools didn't parse setup.cfg at all; the version of setuptools installed by default on Python 3.5 would be one of those versions.

**Comment by fanninpm** [comment]
I have the latest version of `setuptools` installed. Here's the output from `pip show setuptools`:

```
Name: setuptools
Version: 49.1.0
Summary: Easily download, build, install, upgrade, and uninstall Python packages
Home-page: https://github.com/pypa/setuptools
Author: Python Packaging Authority
Author-email: distutils-sig@python.org
License: UNKNOWN
Location: /Users/padraicfanning/Documents/beeware/venv/lib/python3.8/site-packages
Requires: 
Required-by: twine, Sphinx
```

In my local repository, I have created dummy files to block setuptools from writing a package with the wrong name, causing setuptools to fail earlier if it attempts to create `src/UNKNOWN.egg-info`, for example.

**Pull request merged**